nixos/modules/services/hardware/pid-fan-controller.nix

```diff
@@ -0,0 +1,185 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.pid-fan-controller;
+  #floating point option
+  fOpt = (
+    descr:
+    lib.mkOption {
+      type = lib.types.float;
+      description = descr;
+    }
+  );
+  heat_src =
+    { ... }:
```

Comment:
Please use camelCase `heatSrc` or `heatSource`.

Also, a simple attrset should be good here, you can just remove ` { ... }:`. (same for fan below)